app/models/post.rb

```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :comments, class_name: 'Comment', foreign_key: 'post_id'
+  has_many :likes, class_name: 'Like', foreign_key: 'post_id'
+
+  # menthod to update the post counter for a given user
+  after_save :update_post_counter_user
+
+  def update_post_counter_user
+    author.increment!(:posts_counter)
+  end
```

Comment:
```suggestion
  after_save :update_post_counter_user
  private
  def update_post_counter_user
    author.increment!(:posts_counter)
  end
```

### OPTIONAL 

- [ ] Based on the requirements,  It would be better to make the update_post_counter method private, see the suggestion above 👍🏿

Reply:
Thank you for your insightful review. I have made the method private.